OnlineFlightSearchAPITestCases/FlightServicesTests/SearchFlightTests.cs

```diff
@@ -0,0 +1,122 @@
+            var expectedException = new ValidationException(ValidationMessages.InvalidStartLocation);
+            var actualException = Assert.Throws<ValidationException>(() => searchFlightService.SearchFlightDetails(startLocation, "BUD", DateTime.Now.AddDays(1)));
+
+            Assert.Equal(expectedException.Message, actualException.Message);
+        }
+
+        [Theory]
+        [InlineData("XYZ")]
+        public void SearchFlight_IfDestinationIsNotValid_ReturnsValidationException(string endLocation)
+        {
+            IUnityContainer container = Initialize();
+            SearchFlightService searchFlightService = container.Resolve<SearchFlightService>();
+            var expectedException = new ValidationException(ValidationMessages.InvalidDestination);
+            var actualException = Assert.Throws<ValidationException>(() => searchFlightService.SearchFlightDetails("BUD", endLocation, DateTime.Now.AddDays(1)));
+
+            Assert.Equal(expectedException.Message, actualException.Message);
+        }
+
+        [Theory]
+        [InlineData("BUD","BUD")]
+        [InlineData("LTN", "LTN")]
+        public void SearchFlight_IfBothStartAndEndLocationAreSame_ReturnsValidationException(string startLocation,string endLocation)
+        {
+            IUnityContainer container = Initialize();
+            SearchFlightService searchFlightService = container.Resolve<SearchFlightService>();
+            var expectedException = new ValidationException(ValidationMessages.StartandEndLocationCannotBeSame);
+            var actualException = Assert.Throws<ValidationException>(() => searchFlightService.SearchFlightDetails(startLocation, endLocation, DateTime.Now.AddDays(1)));
+
+            Assert.Equal(expectedException.Message, actualException.Message);
+        }
+
+        [Fact]
+        public void SearchFlight_IfDepartureDateIsNotValid_ReturnsValidationException()
+        {
+            IUnityContainer container = Initialize();
+            SearchFlightService searchFlightService = container.Resolve<SearchFlightService>();
+
+            var expectedException = new ValidationException(ValidationMessages.InvalidDepartureDate);
+            var actualException = Assert.Throws<ValidationException>(() => searchFlightService.SearchFlightDetails("BUD", "LTN", DateTime.UtcNow.AddDays(-1)));
+
+            Assert.Equal(expectedException.Message, actualException.Message);
+        }
+
+        [Theory]
+        [InlineData("BUD", "LTN")]
+        public void SearchFlight_IfStartAndEndDestinationAndDepartureDateValid_ReturnsFlightDetails(string startDestination, string endDestination)
+        {
+            IUnityContainer container = Initialize();
+            SearchFlightService searchFlightService = container.Resolve<SearchFlightService>();
+
+            var actualResult = searchFlightService.SearchFlightDetails(startDestination, endDestination, DateTime.UtcNow.AddDays(2));
+
+            Assert.IsType<List<FlightDetail>>(actualResult);
```

Comment:
just the type is good enough? returned flight details are not asserted

Reply:
Validated the resultant Flight List against the input.